packages/next/pages/_document.tsx

```diff
@@ -608,15 +608,15 @@ export class Head extends Component<
             </noscript>
           </>
         )}
-        {children}
-        {process.env.__NEXT_OPTIMIZE_FONTS && (
-          <meta name="next-font-preconnect" />
-        )}
         {head}
         <meta
           name="next-head-count"
           content={React.Children.count(head || []).toString()}
         />
+        {children}
```

Comment:
We will run into the same issue if we have <meta> in _document and have a font in next/head.

The current order of contents is also better from our testing.